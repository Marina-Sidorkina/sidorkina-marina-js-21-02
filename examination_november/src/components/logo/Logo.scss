.logo {
  display: flex;
  height: 70px;
  align-items: center;

  &__img {
    height: 55px;
    object-fit: cover;
    margin-right: 7px;
    margin-bottom: 2px;
  }

  &__text {
    box-sizing: border-box;
    margin-top: 0;
    margin-bottom: 0;
    font-size: 20px;
    font-weight: bold;
  }
}
