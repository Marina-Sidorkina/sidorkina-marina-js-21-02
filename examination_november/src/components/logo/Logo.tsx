import React from 'react';
import './Logo.scss';
import logo from './img/logo.png';

const LOGO_ALT_TEXT = 'Logo image';

const Logo = () => (
  <div className="logo">
    <img className="logo__img" src={logo} alt={LOGO_ALT_TEXT} />
    <p className="logo__text">Delta World</p>
  </div>
);

export default Logo;
