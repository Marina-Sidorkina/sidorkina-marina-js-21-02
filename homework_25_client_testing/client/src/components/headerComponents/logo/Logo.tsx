import React from 'react';
import styles from './Logo.module.scss';
import logo from './img/logo.png';

const Logo = () => (
  <div className={styles.logo}>
    <img className={styles.img} src={logo} alt="Логотип" />
    <p className={styles.text}>Delta World</p>
  </div>
);

export default Logo;
