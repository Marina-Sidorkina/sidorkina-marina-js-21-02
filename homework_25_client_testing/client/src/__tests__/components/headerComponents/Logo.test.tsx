import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import Logo from '../../../components/headerComponents/logo/Logo';

configure({ adapter: new Adapter() });

describe('Logo component test', () => {
  test('should render Header', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper.length).toEqual(1);
  });

  test('should render Logo', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });
});
