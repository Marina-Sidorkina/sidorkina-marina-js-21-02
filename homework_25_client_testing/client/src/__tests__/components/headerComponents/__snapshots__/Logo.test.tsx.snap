// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component test should render Logo 1`] = `
<div
  className="logo"
>
  <img
    alt="Логотип"
    className="img"
    src="logo.png"
  />
  <p
    className="text"
  >
    Delta World
  </p>
</div>
`;
